### ISSUE TITLE
How to generate reproducible results

### ISSUE THREAD

**Issue by mandanraf** [borisdayma/dalle-mini#273]
I want to be able to generate the same set of images each time I run the inference script locally on my machine for reproducibility for the same set of prompts. I am wondering if there is a way possible. 
Thanks.

**Comment by drdaxxy** [comment]
Also keep in mind you'll have to use the same batch size (i.e., number of prompts in the list you pass to `processor()`) and number of GPUs/TPUs (`shard_prng_key()` splits the key by `jax.device_count()`) or you'll get divergent random number sequences.

You can make those sequences independent from batch size by giving each item its own seed (e.g. either patch `generate` to use a `jax.vmap()`-ed version of `jax.random.categorical()`/`jax.random.split()`, or `jax.vmap()` the key parameter on the outside and add an empty dimension to your tokenized prompts)... but due to numerical instability, you won't reliably get the same results.

I suspect this might make results irreproducible between different hardware and/or library/driver versions, but I haven't tested that.

If anyone figures out how to avoid that instability, I'd be very interested!

**Comment by borisdayma** [comment]
Oh I didn't try it but I would expect that if you run the notebook twice with the same seed value you would get the same images, no?
